### PR TITLE
Add Database abstraction and README update

### LINKED ISSUE #26: Add Database abstraction to GCP reference platform
### What problem are you facing?

Currently platform-ref-gcp is lacking behind comparing it to https://github.com/upbound/platform-ref-aws/ and https://github.com/upbound/platform-ref-azure/ where we have Database(Postgresql) abstraction in both cases.

### How could Upbound help solve your problem?

* Add Database abstraction(XRD+Composition) to this platform
* The abstraction should drive https://marketplace.upbound.io/providers/upbound/provider-gcp/v0.13.0/resources/sql.gcp.upbound.io/DatabaseInstance/v1beta1 creation together with auxiliary resources
* k8s cluster that is created by [Cluster abstraction](https://github.com/upbound/platform-ref-gcp/tree/main/cluster) should have connectivity to this database.

### PR DESCRIPTION
**Pull request by ytsarev**
### Description of your changes

* Implement Database abstraction. 
* Use https://cloud.google.com/sql/docs/postgres/configure-private-services-access and https://cloud.google.com/sql/docs/postgres/connect-kubernetes-engine#private-ip for configuring private VPC-based access without cloud SQL proxy sidecar
* Make XRD and Composition naming consistent with the rest of the platform refs
* Update README with a consistent layout. 

Fixes #26 

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested

```
k apply -f examples/cluster-claim.yaml
k apply -f examples/postgres-claim.yaml

 k get claim
NAME                                                                READY   CONNECTION-SECRET          AGE
postgresqlinstance.gcp.platformref.upbound.io/platform-ref-gcp-db   True    platform-ref-gcp-db-conn   129m

NAME                                                          READY   CONNECTION-SECRET   AGE
cluster.gcp.platformref.upbound.io/platform-ref-gcp-cluster   True    cluster-conn        130m

 k view-secret platform-ref-gcp-db-conn
Multiple sub keys found. Specify another argument, one of:
-> privateIP
-> serverCACertificateCert
```

To test the cluster-database private connectivity end-to-end. 
```
#On the target gke cluster

cat no-sidecar-psql.yaml
apiVersion: v1
kind: Pod
metadata:
  name: see-db
  namespace: default
spec:
  containers:
  - name: see-db
    image: postgres:12
    command: ['psql']
    args: ['-c', 'SELECT current_database();']
    env:
    - name: PGDATABASE
      value: postgres
    - name: PGHOST
      value: 10.100.0.3
    - name: PGUSER
      value: platform-ref-gcp-db-r26cb-5qjrf
    - name: PGPASSWORD
      value: <password

k apply -f no-sidecar-psql.yaml

k logs -f see-db
 current_database
------------------
 postgres
(1 row)
```

### CODE REVIEW

**Comment by ytsarev** [comment]
@stevendborrelli please use `upbound/platform-ref-gcp-staging:v0.3.0` for the testing.

**Review by stevendborrelli — APPROVED** [review]
LGTM.

All claims became ready:

```console
kubectl get claim,composite        
NAME                                                          READY   CONNECTION-SECRET   AGE
cluster.gcp.platformref.upbound.io/platform-ref-gcp-cluster   True    cluster-conn        82m

NAME                                                                READY   CONNECTION-SECRET          AGE
postgresqlinstance.gcp.platformref.upbound.io/platform-ref-gcp-db   True    platform-ref-gcp-db-conn   80m

NAME                                                                   READY   COMPOSITION                       AGE
xgke.gcp.platformref.upbound.io/platform-ref-gcp-cluster-xftvn-cp6lf   True    xgke.gcp.platformref.upbound.io   81m

NAME                                                                       READY   COMPOSITION                            AGE
xnetwork.gcp.platformref.upbound.io/platform-ref-gcp-cluster-xftvn-tng6l   True    xnetworks.gcp.platformref.upbound.io   81m

NAME                                                                 READY   COMPOSITION                            AGE
xcluster.gcp.platformref.upbound.io/platform-ref-gcp-cluster-xftvn   True    xclusters.gcp.platformref.upbound.io   81m

NAME                                                                       READY   COMPOSITION                                       AGE
xpostgresqlinstance.gcp.platformref.upbound.io/platform-ref-gcp-db-mf49v   True    xpostgresqlinstances.gcp.platformref.upbound.io   80m

NAME                                                                        READY   COMPOSITION                            AGE
xservices.gcp.platformref.upbound.io/platform-ref-gcp-cluster-xftvn-8qg2l   True    xservices.gcp.platformref.upbound.io   81m
```

File: README.md [hunk 1]
```diff
@@ -105,155 +151,153 @@ Ensure that the following roles are added to your service account:
...
+```
+
+This recommendation is an excerpt from [the official GCP
+documentation](https://cloud.google.com/sql/docs/postgres/configure-private-services-access)
+
+To delete the provisioned resources, you would simply delete the claims again:
 
 ```console
-kubectl -n upbound-system get cluster
+kubectl delete -f examples/cluster-claim.yaml,examples/postgres-claim.yaml
```
Comment (stevendborrelli) [inline README.md:207]: I got caught with hanging resources that required manual deletion (the helm release and SQL user). Perhaps we should document the scripts.

File: examples/configuration.yaml [hunk 2]
```diff
@@ -3,6 +3,6 @@ kind: Configuration
 metadata:
   name: upbound-platform-ref-gcp
 spec:
-  package: xpkg.upbound.io/upbound/platform-ref-gcp:v0.2.1
+  package: xpkg.upbound.io/upbound/platform-ref-gcp:v0.3.0
   packagePullSecrets:
```
[the remark this answers is not in the document]
Comment (ytsarev) [reply examples/configuration.yaml:7]: Yes, it is still around for the custom building/pushing to private repo section

File: examples/postgres-claim.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      name: psqlsecret
+      key: password
+    clusterRef:
+      id: platform-ref-gcp-cluster
+  writeConnectionSecretToRef:
+    name: platform-ref-gcp-db-conn
+---
+apiVersion: v1
+data:
+  password: dXBiMHVuZHIwY2s1ITMxMzM3
```
[the remark this answers is not in the document]
Comment (ytsarev) [reply examples/postgres-claim.yaml:22]: I would leave this step to some future PR which can possibly implement automated App abstraction so we will not expect manual password copy-pasting.

**Pull request merged**